lib/helper/mobile_oauth.dart

```diff
@@ -14,6 +16,8 @@ class MobileOAuth extends CoreOAuth {
   final RequestCode _requestCode;
   final RequestToken _requestToken;
 
+  Completer<String?>? _accessTokenCompleter;
```

Comment:
Thank you @easazade for the PR. Looks good but I've one question, why do we need `_accessTokenCompleter`?

Reply:
since there could multiple parallel calls to `getAccessToken()`, which will result `refreshToken()` being called multiple times which will result in access token being refreshed multiple times. we don't want that. 

the completer is there to make sure no matter how many parallel calls are made to `getAccessToken()` if token is required to be refreshed it will be refreshed only once and one fresh access token will be created while also that single access token be returned for all parties that called `getAccessToken()` in parallel.

Reply:
Thank you for the explanation @easazade , I'll merge this PR.